GenderPayGap.UnitTests/GenderPayGap.WebUI.Tests/Controllers/Account/ChangePasswordController/ChangePasswordControllerTests.cs

```diff
@@ -0,0 +1,245 @@
+
+            // service under test
+            mockUserRepo =
+                new Repositories.UserRepository(
+                    mockDataRepo.Object,
+                    auditLoggerWithMocks);
+        }
+
+        [Test]
+        [Description("POST: Password is updated as expected")]
+        public async Task POST_Password_Is_Updated_As_Expected()
+        {
+            // Arrange
+            User user = new UserBuilder().WithPassword("password").Build();
+            
+            var requestFormValues = new Dictionary<string, StringValues>();
+            requestFormValues.Add("GovUk_Text_CurrentPassword", "password");
+            requestFormValues.Add("GovUk_Text_NewPassword", "NewPassword1");
+            requestFormValues.Add("GovUk_Text_ConfirmNewPassword", "NewPassword1");
+
+            var controller = new ControllerBuilder<ChangePasswordNewController>()
+                .WithLoggedInUser(user)
+                .WithRequestFormValues(requestFormValues)
+                .WithDatabaseObjects(user)
+                .WithMockUriHelper()
+                .Build();
+
+            // Act
+            controller.ChangePasswordPost(new ChangePasswordNewViewModel()).Wait();
+            
+            // Assert
+            bool isExpectedPassword = mockUserRepo.CheckPassword(user, "NewPassword1");
+            Assert.IsTrue(isExpectedPassword);
+        }
+        
+        [Test]
+        [Description("POST: Email is sent when password is successfully updated")]
+        public async Task POST_Email_Is_Sent_When_Password_Is_Successfully_Updated()
+        {
+            // Arrange
+            User user = new UserBuilder().WithPassword("password").Build();
+            
+            var requestFormValues = new Dictionary<string, StringValues>();
+            requestFormValues.Add("GovUk_Text_CurrentPassword", "password");
+            requestFormValues.Add("GovUk_Text_NewPassword", "NewPassword1");
+            requestFormValues.Add("GovUk_Text_ConfirmNewPassword", "NewPassword1");
+            
+            var controllerBuilder = new ControllerBuilder<ChangePasswordNewController>();
+            var controller = controllerBuilder
+                .WithLoggedInUser(user)
+                .WithRequestFormValues(requestFormValues)
+                .WithDatabaseObjects(user)
+                .WithMockUriHelper()
+                .Build();
+
+            // Act
+            controller.ChangePasswordPost(new ChangePasswordNewViewModel()).Wait();
+            
+            // Assert
+            // Assert that exactly one email is sent
+            Assert.That(controllerBuilder.EmailsSent.Count == 1);
+            
+            NotifyEmail userEmail = controllerBuilder.EmailsSent.FirstOrDefault();
+            // Assert that the email sent has the correct email address and template
+            Assert.NotNull(userEmail);
+            Assert.AreEqual(userEmail.EmailAddress, user.EmailAddress);

```

Comment:
A couple of really minor points about how you're using the Assertion library:

If we use it correctly, Assert.* can give us reasonably helpful error messages, e.g.
```
Test Failure : TestClassName.TestMethodName
  Expected: user1@example.com
  But was:  user2@example.com
```

To make this work, we need to:
* Use `Assert.AreEqual()` rather than `Assert.That()`
* Make sure we get the parameters the right way round (expected result first, then actual result) - otherwise the message will look confusing!

In this case, the parameters are the wrong way round

Reply:
Ah ok, I didn't realise that the choice made a difference to the error message, but that makes sense now I think about it!